core/app/[locale]/(default)/(faceted)/search/page.tsx

```diff
@@ -1,64 +1,76 @@
 import { getFormatter, getTranslations } from 'next-intl/server';
+import { createSearchParamsCache } from 'nuqs/server';
+import { cache } from 'react';
 
 import { Breadcrumb } from '@/vibes/soul/primitives/breadcrumbs';
 import { CursorPaginationInfo } from '@/vibes/soul/primitives/cursor-pagination';
 import { ListProduct } from '@/vibes/soul/primitives/products-list';
 import { ProductsListSection } from '@/vibes/soul/sections/products-list-section';
+import { getFilterParsers } from '@/vibes/soul/sections/products-list-section/filter-parsers';
 import { Filter } from '@/vibes/soul/sections/products-list-section/filters-panel';
 import { Option as SortOption } from '@/vibes/soul/sections/products-list-section/sorting';
 import { facetsTransformer } from '~/data-transformers/facets-transformer';
 import { pricesTransformer } from '~/data-transformers/prices-transformer';
 
-import { createFiltersSearchParamCache } from '../create-filters-search-params-cache';
 import { fetchFacetedSearch } from '../fetch-faceted-search';
 
-type SearchParams = Promise<Record<string, string | string[] | undefined>>;
+interface Props {
+  searchParams: Promise<Record<string, string | string[] | undefined>>;
```

Comment:
@migueloller Should we use `nuqs` `SearchParams` here?

Reply:
No because these are the search params Next.js passes to us _before_ we've parsed them with nuqs. I parse them in utility functions using `createSearchParamsCache`